File: Java/Creational/Builder/Example1/Main/src/Product.java
import java.util.ArrayList;
import java.util.List;

public class Product {
    private List<String> parts = new ArrayList<>();

    public void add(String part) {
        parts.add(part);
    }

    public void show() {
        System.out.println("Product parts:");
        for (String part : parts) {
            System.out.println("- " + part);
        }
    }
}

File: TheRayCode/PHP/Creational/Builder/Example/Product.php
<?php

class Product
{
    private array $parts = [];

    // Add a part to the product
    public function addPart(string $part): void
    {
        $this->parts[] = $part;
    }

    // Show the parts that have been added to the product
    public function show(): void
    {
        echo "Product Parts:\n";
        foreach ($this->parts as $part) {
            echo "- $part\n";
        }
    }

}